Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -2651,23 +2649,50 @@ internal static async Task ListenNotificationsAsync_Test1(IMinioClient minio)
 
             var eventsList = new List<EventType> { EventType.BucketCreatedAll };
 
-            var events = minio.ListenNotificationsAsync(eventsList);
+            var listenArgs = new ListenBucketNotificationsArgs()
+                .WithEvents(eventsList);
+            var events = minio.ListenNotifications(listenArgs);
             var subscription = events.Subscribe(
                 received.Add,
-                ex => Console.WriteLine($"OnError: {ex}"),
-                () => Console.WriteLine("Stopped listening for bucket notifications\n"));
-
-            // Ensure the subscription is established
-            await Task.Delay(1000).ConfigureAwait(false);
-
+                _ => { },
+                () => { }
+            );
             // Trigger the event by creating a new bucket
-            var isBucketCreated1 = await CreateBucket_Tester(minio, bucketName).ConfigureAwait(false);
+            _ = await CreateBucket_Tester(minio, bucketName).ConfigureAwait(false);
 
             var eventDetected = false;
             for (var attempt = 0; attempt < 20; attempt++)
-            {
                 if (received.Count > 0)
                 {
+                    // Check if there is any unexpected error returned
+                    // and captured in the receivedJson list, like
+                    // "NotImplemented" api error. If so, we throw an exception
+                    // and skip running this test
+                    if (received.Count > 1 &&
+                        received[1].Json.StartsWith("<Error><Code>", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Although the attribute is called "json",
+                        // returned data in list "received" is in xml
+                        // format and it is an error.Here, we convert xml
+                        // into json format.
+                        var receivedJson = XmlStrToJsonStr(received[1].Json);
+
+                        // Cleanup the "Error" key encapsulating "receivedJson"
+                        // data. This is required to match and convert json data
+                        // "receivedJson" into class "ErrorResponse"
+                        var len = "{'Error':".Length;
+                        var trimmedFront = receivedJson[len..];
+                        var trimmedFull = trimmedFront[..^1];
+
+                        var err = JsonSerializer.Deserialize<ErrorResponse>(trimmedFull);
```

Comment:
Why not parse as XML instead? This kind of "trimming" the JSON is error-prone. If I read it correctly you assume that it always starts with `{'Error':` and it always ends with `}`. JSON that would have inserted spaces would have been correct JSON too, so better use a regular JSON deserializer and fetch the `Error` object from it.

Even though "it just a unit-test", we should keep code clean.

Reply:
Actually it turned out to be the whole logic here was wrong and this part of the code was completely unnecessary.
So, I removed the part of the code, including the trimming, that would never be executed.
Now, it looks much cleaner.
I also modified the next test, `ListenBucketNotificationAsync_Test1`, which also had the same logic.
This is actually should have been addressed in a separate issue and  PR, but since  the work is already done, I did not remove it. (Btw, the same cleanup needs to be done for methods; `ListenBucketNotificationAsync_Test2` and `ListenBucketNotificationAsync_Test3`, but I'd address them in a separate issue with reference to this PR).

Reply:
@ramondeklein 
Could you please check the code again?

Reply:
Lint changes are in its way...

Reply:
When I run `dotnet regitlint` it did not find any lint problem... looking into it

Reply:
for some reason running `dotnet regitlint` locally did not  report any problem but there were some offending changes which were causing the build test format check to fail.
Adding the `--print-diff` option in the `minio-dotnet.yml --fail-on-diff` helped to see those missing lint changes. The last commit resolved the issue and all tests passed.

@ramondeklein 
Could you re-review the changes?